src/main/java/org/genomenexus/vep_wrapper/ProcessSurveyor.java

```diff
@@ -0,0 +1,98 @@
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+import java.io.StringWriter;
+import java.sql.Timestamp;
+import java.util.*;
+
+public class ProcessSurveyor {
+
+    private static ArrayList<String> commandElements;
+    private static ProcessBuilder psProcessBuilder;
+
+    static {
+        // initialize commandElements
+        commandElements = new ArrayList<String>();
+        commandElements.add("ps");
+        commandElements.add("axo");
+        commandElements.add("pid,ppid,state,comm");
+        // initialize processBuilder
+        psProcessBuilder = new ProcessBuilder(commandElements);
+        psProcessBuilder.redirectErrorStream(false);
+        if (SystemProcessManager.getDevNull() != null) {
+            psProcessBuilder.redirectError(ProcessBuilder.Redirect.to(SystemProcessManager.getDevNull()));
+        }
+    }
+
+    public static ProcessSurveyItem parsePsOutputLine(String line) {
+        // each output line from the ps command will have 4 fields, such as these example lines:
+        //  PID  PPID S COMMAND
+        //    1     1 S java
+        //   46     1 R perl
+        //   48     1 R ps
+        // the leading spaces are not guaranteed to be present
+        if (line == null || line.length() < 8) {
+            return null; // not enough content in line
+        }
+        String[] fields = line.split("[ ]+");
+        if (fields.length < 4) {
+            return null; // not enough fields to parse
+        }
+        int firstFieldIndex = 0;
+        if (fields[0].length() == 0) {
+            // skip empty field due to leading spaces
+            firstFieldIndex = firstFieldIndex + 1;
+        }
+        int processId = 0;
+        try {
+            processId = Integer.parseInt(fields[firstFieldIndex]);
+        } catch (NumberFormatException e) {
+            return null; // first column is not a number .. such as in the header line
+        }
+        int parentProcessId = 0;
+        try {
+            parentProcessId = Integer.parseInt(fields[firstFieldIndex + 1]);
+        } catch (NumberFormatException e) {
+            return null; // first column is not a number .. such as in the header line
+        }
+        if (fields[firstFieldIndex + 2].length() != 1) {
+            return null;
+        }
+        char stateCode = fields[firstFieldIndex + 2].charAt(0);
+        if (fields[firstFieldIndex + 3].length() == 0) {
+            return null;
+        }
+        return new ProcessSurveyItem(processId, parentProcessId, stateCode, fields[firstFieldIndex + 3]);
+    }
+
+    public static List<ProcessSurveyItem> getProcessSurvey() {
+        ArrayList<ProcessSurveyItem> processSurvey = new ArrayList<ProcessSurveyItem>();
+        try {
+            Process psProcess = psProcessBuilder.start();
+            BufferedReader in = new BufferedReader(new InputStreamReader(psProcess.getInputStream()));
+            while (true) {
+                String line = in.readLine();
```

Comment:
isn't it equivalent to use:

```
while ((line = in.readLine()) != null)
...
```

?

I am hesitant to use `true` in the while statement

Reply:
i suppose the item being parsed would also need to be in the while statement check as well

Reply:
fixed as you suggested